vllm/config.py

```diff
@@ -1191,21 +1191,8 @@ class DeviceConfig:
     def __init__(self, device: str = "auto") -> None:
         if device == "auto":
             # Automated device type detection
-            if current_platform.is_cuda_alike():
-                self.device_type = "cuda"
-            elif current_platform.is_neuron():
-                self.device_type = "neuron"
-            elif current_platform.is_hpu():
-                self.device_type = "hpu"
-            elif current_platform.is_openvino():
-                self.device_type = "openvino"
-            elif current_platform.is_tpu():
-                self.device_type = "tpu"
-            elif current_platform.is_cpu():
-                self.device_type = "cpu"
-            elif current_platform.is_xpu():
-                self.device_type = "xpu"
-            else:
+            self.device_type = current_platform.device_type
```

Comment:
I think this will raise exception (attribute error) directly.

Reply:
need to set it in `UnspecifiedPlatform` too

Reply:
Could you tell me in what cases `UnspecifiedPlatform` would be called? I cannot confirm the `device_type` of that.

Reply:
if we install a cpu version pytorch in gpu environment (like arm64 chips, GH200), i think this will happen.

Reply:
I'd like to keep the original error message `raise RuntimeError("Failed to infer device type")` , but i'm afraid this will become an attribute after this pr.

Reply:
Thanks for your explanation, I got your point. Sorry for missing this case.